### ISSUE TITLE
Error in optim: initial value in 'vmmin' is not finite

### ISSUE THREAD

**Issue by krpaulson** [taylorokonek/pssst#12]
Hi Taylor,

I'm getting the following error for Rwanda 2000 DHS:

```
> df_formatted <- pssst::format_dhs(
+  df = df, survey_year = survey_year, period_boundaries = period_boundaries,
+  right_censor_time = 60, intervals = c("6-18")
+ )
> fit_loglogistic <- pssst::surv_synthetic(
+   df = df_formatted, household = "household", cluster = "cluster", strata = "strata",
+   weights = "weights", numerical_grad = TRUE, dist = "loglogistic",
+   init_vals = c(rep(-1.5, length(period_boundaries)-1), rep(15, length(period_boundaries)-1))
+ )
fitting model
Error in optim(par = init_vals, fn = optim_fn, data = df[, c("I_i", "A_i",  : 
  initial value in 'vmmin' is not finite
```

I spent a little bit of time attempting to debug on my own but didn't get anywhere. This is not urgent but if you have a moment to take a look, a more informative error message would be super nice if this is a potentially recurring issue. I presume it's something atypical with the input data but it's not obvious to me what that is. Let me know if you'd like me to email you the `df_formatted` data here.

Thank you!!

**Comment by taylorokonek** [comment]
This is one of those issues that occurs in optim... have you tried different starting values (init_vals)? There should be an option for that in the code already, I would try that first to see if it works

**Comment by krpaulson** [comment]
Yeah I've set `init_vals = c(rep(-1.5, length(period_boundaries)-1), rep(15, length(period_boundaries)-1))` based on typical values across ~10 countries and it seems to usually work. Just in case, I just tested a few other values for this survey and still got the same error.

**Comment by taylorokonek** [comment]
What I would suggest doing is taking those starting values, going inside the function, and fitting the log-likelihood calculation just once (one iteration, no optim), and then looking to see if any are returning weird values (Infinity, or -Infinity, basically). Then you can identify which observations are causing the problem and we can go from there

**Comment by krpaulson** [comment]
Yeah, okay. I had the same thought but it took me a minute to figure out how to do that with the Rcpp stuff. What I found out is that there was a person whose exact age was "199" who got assigned `t_0i = 3.3` and `t_1i = 3.3` initially, but this was changed to `t_0i = 3.3` and `t_1i = 2` here: https://github.com/taylorokonek/pssst/blob/main/R/format_dhs.R#L291-L311. Since 3.3>2 this gave us -Inf within `optim_fn`, hence the error.

I think next steps are: (1) figure out if that section within `format_dhs` is doing what we want it to be doing, (2) perhaps add a check somewhere that `t_0i < t_1i` to give an informative error message. What do you think?

**Comment by taylorokonek** [comment]
Okay cool, thanks for identifying the problem! I think it's likely a `format_dhs` issue. I think next steps are (1) adding in an error message at the beginning of the likelihood function to check that t_0i <= t_1i (like you noted, except with the equal option as well since that's how exactly observed observations are recorded), and then looking into `format_dhs` again....
